File: aemet/Aemet-Java-IvanRC/src/main/java/models/Medicion.java
package models;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.time.LocalTime;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class Medicion {
    String poblacion;
    String provincia;
    Double temperaturaMax;
    LocalTime horaTempMax;
    Double temperaturaMin;
    LocalTime horaTempMin;
    Double precipitacion;
    LocalDate dia;

    public LocalTime getTempMax(){
        return horaTempMax;
    }

    public Double getTemperaturaMedia(){
        return (temperaturaMax+temperaturaMin)/2;
    }
}

File: aemet/Aemet-kotlin-IvanRC/src/main/kotlin/model/Medicion.kt
package model

import java.time.LocalDate
import java.time.LocalTime

data class Medicion (
    val poblacion: String,
    val provincia: String,
    val temperaturaMax: Double,
    val horaTempMax: LocalTime,
    val temperaturaMin: Double,
    val horaTempMin: LocalTime,
    val precipitacion: Double,
    val dia: LocalDate
)